Source/KSCrash/Recording/Monitors/KSCrashMonitor_CPPException.cpp

```diff
@@ -82,12 +94,14 @@ extern "C"
 
     void __cxa_throw(void* thrown_exception, std::type_info* tinfo, void (*dest)(void*))
     {
-        if(g_captureNextStackTrace)
+        static cxa_throw_type orig_cxa_throw = NULL;
+        if (g_cxaSwapEnabled == false)
         {
-            kssc_initSelfThread(&g_stackCursor, 1);
+            if(g_captureNextStackTrace)
+            {
+                kssc_initSelfThread(&g_stackCursor, 1);
```

Comment:
This is in multiple places now, and also has a comment about a temporary hack in the other place. Let's move this logic to a central function so that any later fixes don't cause regressions.